doc/api/modules.markdown

```diff
@@ -429,6 +429,18 @@ of absolute paths, then io.js will search those paths for modules if they
 are not found elsewhere.  (Note: On Windows, `NODE_PATH` is delimited by
 semicolons instead of colons.)
 
+`NODE_PATH` was originally created to support loading modules from
+varying paths before the current
+[module resolution](https://iojs.org/api/modules.html) algorithm was
+frozen.
+
+`NODE_PATH` is still supported, but is less necessary now that the io.js
+ecosystem has settled on a convention for locating dependent modules.
+Sometimes deployments that rely on `NODE_PATH` show surprising behavior
+when people are unaware that `NODE_PATH` must be set.  Sometimes a
+module's dependencies change, causing a different version (or even a
+different module) to be loaded as the `NODE_PATH` is searched.
```

Comment:
We go on in the document to suggest using NODE_PATH to make global modules require-able in the REPL – we should probably nix that.

I'd also suggest bolding the line below, "You are highly encouraged to place your dependencies locally in `node_modules` folders."